tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -143,32 +143,50 @@ def _ValidateTranslation(self):
         print('Invalid translation compliance string', translation_body.data)
         return False
 
+    # TODO(charbull): refactor this while refactoring the class
+    # validate fields
+    entity_type_str = str(self.entity['type'])
+    type_parse = entity_type_str.split('/')
+    namespace = type_parse[0]
+    entity_type = type_parse[1]
+    entity_type = self.universe.GetEntityType(namespace, entity_type)
+    all_fields_dict = entity_type.GetAllFields()
+
     # iterate through each translation device key and determine its form
-    for device_name in translation_body.keys():
+    for field_name in translation_body.keys():
       # check if keys are UDMI compliant
-      if isinstance(translation_body[device_name].data, str):
+      if isinstance(translation_body[field_name].data, str):
         continue
 
-      device_map = translation_body[device_name]
+      #check if the field_name is on the type
+      # TODO(charbull), the key in the dictionary all_fields_dict starts with `/`, needs to be cleaned
+      # pop the field out
+      key_field_name = '/'+field_name.data
+      opt_wrapper_field = all_fields_dict.pop(key_field_name, None)
+      if opt_wrapper_field is None: #an extra field that should not be here
+        print('Invalid extra field present:', field_name)
+        return False
+
+      translation_map = translation_body[field_name]
       valid_units = self.universe.unit_universe.GetUnitsMap('').keys()
       valid_states = self.universe.state_universe.GetStatesMap('').keys()
 
       # three remaining possibilities for translation format
-      if self.unit_values in device_map.keys():
-        unit_values_map = device_map[self.unit_values]
+      if self.unit_values in translation_map.keys():
+        unit_values_map = translation_map[self.unit_values]
         for unit in unit_values_map.keys():
           if unit not in valid_units:
             print('Invalid translation unit', unit)
             return False
-      elif self.states in device_map.keys():
-        states_map = device_map[self.states]
+      elif self.states in translation_map.keys():
+        states_map = translation_map[self.states]
```

Comment:
this can be stricter because states are dictated by field.

Reply:
how do we extract that? with string manipulation?
I prefer to do this in the universe config in a different PR?